probe-rs/src/probe/daplink/tools.rs

```diff
@@ -180,7 +180,11 @@ pub fn open_device_from_selector(
             };
 
             let timeout = Duration::from_millis(100);
-            let language = handle.read_languages(timeout)?[0];
+            let language = match handle.read_languages(timeout)?.get(0) {
```

Comment:
Same issue as in the comment above, I believe :)

Reply:
So I adjusted the `sn_str` reading so that it is `None` if the language cannot be read instead of skipping the device entirely. But if I am not mistaken, the overall outcome is the same, because open_v2_device will reject the device anyway?